text/0715-construct-upgrade-support.md

```diff
@@ -0,0 +1,423 @@
+
+### Why should I use this feature?
+
+CDK will maintain old constructs like `aws-cdk-lib.aws-dynamodb.Table` and `aws-cdk-lib.aws-ec2.Vpc`
+with security fixes only. If CDK users want to use the new and latest features, we highly
+recommend users to migrate to the V2 constructs or modules for long term maintainability.
+This feature provides a safe and semi-automated method for the migration.
+
+### What if I incorrectly migrate a stack?
+
+The CLI offers mechanisms to prevent accidental bugs, such as dry runs and a
+number of validations that will be performed with the CLI command. Nevertheless,
+mistakes can happen.
+
+If you're migrating using In-Place migration approach, then
+CloudFormation stack refactoring feature will perform additional validations and
+rollback automatically. If you accidentally migrate a stack, you can run the CLI with
+inverse mapping file `refactor.json` to bring the resources back to their original state.
+
+If you're migrating using Retain-Remove-Import migration approach, then CloudFormation
+deployment will rollback the stack for you automatically if things go wrong. If you
+want to switch back to V1 construct from V2 cosntruct, you can do this by retaining
+V2 resource, remove it from stack and import it using V1 CDK code.
+
+## Internal FAQ
+
+### Why are we doing this?
+
+A number of CDK customers have requested this feature. CDK construct upgrade support
+will show customer obsession by offering customers a safe and monitored method
+for migrations and it will increase the adoption rate for new constructs and modules
+as CDK will gradually reduce the level of supports for the outdated constructs and modules
+to eventually Keep-The-Light-On mode.
+
+### Is this a breaking change?
+
+No. Depending on the migration strategies used, the CLI will either update the resource in place
+using CloudForamtion/CDK stack refactoring feature or use CDK import feature to retain existing
+resource, remove it from stack, and import it back into stack. Also, this
+feature will initially be launched in experimental mode, and users must
+acknowledge this by passing the `--unstable=migrate` flag.
+
+### What is the high-level project plan?
+
+#### Milestone 1
+
+Milestone 1 is the initial, experimental release of CDK miconstruct upgrade feature. This goal of this
+milestone is to combine s guided documentation, automate validations, and deployment safeguards
+to ensure a controlled transition. We expect users to follow the documentation and do the
+code migration process themselves.
+
+##### Phase 1 (dry-run)
+
+This phase will ensure through the automated migration tool that user’s existing CDK stack and
+environment area ready for migration, flagging warnings and blockers like CloudFormation drift.
+Additionally, validate that the migrated code is safe to deploy without causing downtime or replacement
+through CloudFromation template comparison and change set analysis.
+
+High-level tasks:
```

Comment:
Seems like these validations are the main added value of this feature. Can you please elaborate on them a bit? give examples for each? 

Also - have you considered putting them inside `cdk diff`? at the end of the day, they are all based on template comparison, which is what `cdk diff` does already.